canary/webrtc-c/src/CanaryWebrtc.cpp

```diff
@@ -269,19 +268,25 @@ VOID sendCustomFrames(Canary::PPeer pPeer, MEDIA_STREAM_TRACK_KIND kind, UINT64
     UINT32 hexStrLen = 0;
     UINT32 actualFrameSize = 0;
     UINT32 frameSizeWithoutNalu = 0;
-    // This is the actual frame size that includes the metadata and the actual frame data
-    actualFrameSize = CANARY_METADATA_SIZE + ((dataRate / 8) / frameRate);
-    frameSizeWithoutNalu = actualFrameSize - ANNEX_B_NALU_SIZE;
-
+    UINT32 minFrameSize = CANARY_METADATA_SIZE + ((dataRate / 8) / frameRate);
+    UINT32 maxFrameSize = (CANARY_METADATA_SIZE + ((dataRate / 8) / frameRate)) * 2;
     PBYTE canaryFrameData = NULL;
-    canaryFrameData = (PBYTE) MEMALLOC(actualFrameSize);
+    
+    canaryFrameData = (PBYTE) MEMALLOC(maxFrameSize);
 
     // We allocate a bigger buffer to accomodate the hex encoded string
-    frame.frameData = (PBYTE) MEMALLOC(frameSizeWithoutNalu * 2 + 1 + ANNEX_B_NALU_SIZE);
+    frame.frameData = (PBYTE) MEMALLOC(maxFrameSize * 3 + 1 + ANNEX_B_NALU_SIZE);
```

Comment:
Did you test H264 with this?

Reply:
Yes, I ran the 90 seconds test with these changes and it works for H264 and H265 both